src/collections/Table/Table.js

```diff
@@ -1,23 +1,13 @@
 import _ from 'lodash';
 import React, {Children, Component, PropTypes} from 'react';
 import classNames from 'classnames';
-import TableColumn from './TableColumn';
+import {childrenOfTypesOnly} from '../../utils/customPropTypes';
 import META from 'src/utils/Meta';
 
 export default class Table extends Component {
   static propTypes = {
     basic: PropTypes.bool,
-    children: (props, propName, componentName) => {
-      let isValid = true;
-      Children.forEach(props.children, (child) => {
-        if (child.type !== TableColumn) {
-          isValid = false;
-        }
-      });
-      if (!isValid) {
-        return new Error('`Table` must only have `TableColumn` children.');
-      }
-    },
+    children: childrenOfTypesOnly(['TableColumn']),
```

Comment:
Our previous conversation about this type check only working on children was misguided.  After finding [this React issue](https://github.com/facebook/react/issues/2979), it seems the reason `instanceOf` doesn't work is because it is strictly doing `x instanceof y`.  However, when creating a class that extends `React.Component` instanceof checks apparently aren't going to work for things like, `TableColumn`.

**TL;DR**
I think a more appropriate name for this util is something like `componentTypes()` or `componentNames()` and drop the `children` reference.  Since, this is in fact what we are checking and the util will work for all props, not just `children`.